v-next/hardhat/src/internal/core/hre.ts

```diff
@@ -205,6 +205,7 @@ async function resolveUserConfig(
     plugins: sortedPlugins,
     tasks: config.tasks ?? [],
     paths: resolvePaths(projectRoot, configPath, config.paths),
+    defaultNetwork: config.defaultNetwork ?? "hardhat",
```

Comment:
This needs confirmed. I suspect this is a fix for a bug whereby the `defaultNetwork` from config was being ignored and always set to `hardhat`.

Reply:
correct